reviews/models/mixins.py

```diff
@@ -171,9 +175,33 @@ def save_changes(self, ev):
         now = self.action.date_created if self.action is not None else timezone.now()
         should_publish = self.reviewable.in_public_reviews_state
         if should_publish and not self.reviewable.is_published:
-            self.reviewable.is_published = True
+            if not (self.reviewable.node.preprint_file and self.reviewable.node.preprint_file.node == self.reviewable.node):
```

Comment:
This is fine for now, but we should generally try not to assume that all reviewables are preprints. At some point (maybe post-MVP) we should add some `NotImplemented` methods on `ReviewableMixin` like `valid_reviewable_to_publish` and `publish_reviewable`, or something like that. And modify `PreprintService.set_published` so we can use it in `publish_reviewable` instead of repeating logic here.